src/lib.rs

```diff
@@ -69,6 +70,104 @@ impl Mapping {
     }
 }
 
+/// Monotonically-increasing generator of identifiers.
+pub struct IdGenerator {
+    last_id: AtomicUsize,
+}
+
+impl IdGenerator {
+    /// Constructs a new generator that starts at the given value.
+    fn new(start_value: u64) -> Self {
+        IdGenerator { last_id: AtomicUsize::new(start_value as usize) }
+    }
+
+    /// Obtains a new identifier.
+    pub fn next(&self) -> u64 {
+        let id = self.last_id.fetch_add(1, Ordering::SeqCst);
```

Comment:
AcqRel is sufficient here. SeqCst is super expensive on some architectures.

Reply:
Hadn't read the various options before. Seems reasonable; done.